File: Task 4/Java/app/src/main/java/com/example/fragmentexample/MainActivity.java
package com.example.fragmentexample;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Dynamically add the fragment to the Activity
        if (savedInstanceState == null) {
            Fragment myFragment = new MyFragment();
            FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
            transaction.replace(R.id.fragment_container, myFragment);
            transaction.commit();
        }
    }
}

File: Task 4/Kotlin/app/src/main/java/com/example/fragmentexample/MainActivity.kt
package com.example.fragmentexample

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btn_Fragment1: Button = findViewById(R.id.btn_fragment1)
        val btn_Fragment2: Button = findViewById(R.id.btn_fragment2)

        val fragment1 = Fragment1()
        val fragment2 = Fragment2()

        supportFragmentManager.beginTransaction().apply {
            replace(R.id.fl_fragment, fragment1)
            commit()
        }

        btn_Fragment1.setOnClickListener {
            supportFragmentManager.beginTransaction().apply {
                replace(R.id.fl_fragment, fragment1)
                addToBackStack(null)
                commit()
            }
        }

        btn_Fragment2.setOnClickListener {
            supportFragmentManager.beginTransaction().apply {
                replace(R.id.fl_fragment, fragment2)
                addToBackStack(null)
                commit()
            }
        }
    }
}